Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs

```diff
@@ -82,12 +96,190 @@ sendPaymentReminderToDriver Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId
       driver <- B.runInReplica $ QPerson.findById (cast driverFee.driverId) >>= fromMaybeM (PersonDoesNotExist driverFee.driverId.getId)
       -- driver <- QPerson.findById (cast driverFee.driverId) >>= fromMaybeM (PersonDoesNotExist driverFee.driverId.getId)
       transporterConfig <- SCT.findByMerchantId driver.merchantId >>= fromMaybeM (TransporterConfigNotFound driver.merchantId.getId)
-      ReSchedule <$> getRescheduledTime transporterConfig
+      ReSchedule <$> getRescheduledTime transporterConfig.driverPaymentReminderInterval
 
-unsubscribeDriverForPaymentOverdue ::
+cacheDriverPlan :: (CacheFlow m r) => Id Driver -> UTCTime -> Plan -> m ()
+cacheDriverPlan driverId time driverPlan = Hedis.setExp (makeDriverPlanKey driverId time) driverPlan 86399 -- Hedis.withCrossAppRedis $
+
+makeDriverPlanKey :: Id Driver -> UTCTime -> Text
+makeDriverPlanKey id day = "driver-offer:CachedQueries:DriverPlan:Date" <> show day <> "DriverId-" <> id.getId
+
+getDriverPlanCache :: CacheFlow m r => UTCTime -> Id Driver -> m (Maybe Plan)
+getDriverPlanCache time driverId = Hedis.get (makeDriverPlanKey driverId time)
+
+calculateDriverFeeForDrivers ::
   ( CacheFlow m r,
     EsqDBFlow m r,
-    Esq.EsqDBReplicaFlow m r
+    EncFlow m r
+  ) =>
+  Job 'CalculateDriverFees ->
+  m ExecutionResult
+calculateDriverFeeForDrivers Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId) do
+  -- handle 1st time
+  let jobData = jobInfo.jobData
+      merchantId = jobData.merchantId
+      startTime = jobData.startTime
+      endTime = jobData.endTime
+  firstJobOfWindow <- getDriverFeeCalcJobFlagKey startTime endTime merchantId
+  when (firstJobOfWindow == Just True) $ do
+    driverFees <- findFeesInRangeWithStatus startTime endTime ONGOING Nothing
+    for_ driverFees $ \driverFee -> do
+      mbDriverPlan <- findByDriverId (cast driverFee.driverId)
+      plan <- getPlan mbDriverPlan merchantId
+      cacheDriverPlan driverFee.driverId endTime plan
```

Comment:
Sugession for DB optimization:

instead of each individual plan, lets divide these `m` plans into `n` batches of batchSize `x` and store in redis as:
```
driversPlanToCalculate-Batch-1 : [(dp1,df1) , (dp2, df2), (dp3,df3)]
driversPlanToCalculate-Batch-2 : [..., ...., ....]
...
...
driversPlanToCalculate-Batch-n : [... , ... , (dpm, dfm)]
```
along with a batchNum counter key - `currentCalulationBatchNum: 1` (init value)
then we would not have to query the DB again for plans `driverFeeCalculatorBatchSize` as currently doing in line 135. just incr the `currentCalulationBatchNum` key before rescheduling and fetch the batch to process from redis.

Reply:
Yup, we already planned for this, will go in next iteration, this is happy flow code